plugins/woocommerce/src/Internal/Orders/OrderActionsRestController.php

```diff
@@ -0,0 +1,128 @@
+		register_rest_route(
+			$this->route_namespace,
+			'/orders/(?P<id>[\d]+)/actions/send_order_details',
+			array(
+				array(
+					'methods'             => \WP_REST_Server::CREATABLE,
+					'callback'            => fn( $request ) => $this->run( $request, 'send_order_details' ),
+					'permission_callback' => fn( $request ) => $this->check_permissions( $request ),
+					'args'                => $this->get_args_for_order_actions(),
+					'schema'              => $this->get_schema_for_order_actions(),
+				),
+			)
+		);
+	}
+
+	/**
+	 * Permission check for REST API endpoint.
+	 *
+	 * @param WP_REST_Request $request The request for which the permission is checked.
+	 * @return bool|WP_Error True if the current user has the capability, otherwise a WP_Error object.
+	 */
+	private function check_permissions( WP_REST_Request $request ) {
+		$order_id = $request->get_param( 'id' );
+		$order    = wc_get_order( $order_id );
+
+		if ( ! $order ) {
+			return new WP_Error( 'woocommerce_rest_not_found', __( 'Order not found', 'woocommerce' ), array( 'status' => 404 ) );
+		}
+
+		return $this->check_permission( $request, 'read_shop_order', $order_id );
+	}
+
+	/**
+	 * Get the accepted arguments for the POST request.
+	 *
+	 * @return array[]
+	 */
+	private function get_args_for_order_actions(): array {
+		return array(
+			'id' => array(
+				'description' => __( 'Unique identifier of the order.', 'woocommerce' ),
+				'type'        => 'integer',
+				'context'     => array( 'view', 'edit' ),
+				'readonly'    => true,
+			),
+		);
+	}
+
+	/**
+	 * Get the schema for both the GET and the POST requests.
+	 *
+	 * @return array[]
+	 */
+	private function get_schema_for_order_actions(): array {
+		$schema['properties'] = array(
+			'message' => array(
+				'description' => __( 'A message indication whether the email was sent.', 'woocommerce' ),
+				'type'        => 'string',
+				'context'     => array( 'view', 'edit' ),
+				'readonly'    => true,
+			),
+		);
+		return $schema;
+	}
+
+	/**
+	 * Handle the POST /orders/{id}/actions/send_order_details.
+	 *
+	 * @param WP_REST_Request $request The received request.
+	 * @return array|WP_Error Request response or an error.
+	 */
+	public function send_order_details( WP_REST_Request $request ) {
+		$order_id = $request->get_param( 'id' );
+		$order    = wc_get_order( $order_id );
+		if ( ! $order ) {
+			return new WP_Error( 'invalid_order', __( 'Invalid order ID.', 'woocommerce' ), array( 'status' => 404 ) );
+		}
+
+		// phpcs:disable WooCommerce.Commenting.CommentHooks.MissingSinceComment
+		/** This action is documented in includes/admin/meta-boxes/class-wc-meta-box-order-actions.php */
+		do_action( 'woocommerce_before_resend_order_emails', $order, 'customer_invoice' );
+
+		WC()->payment_gateways();
+		WC()->shipping();
+		WC()->mailer()->customer_invoice( $order );
+
+		$order->add_order_note( __( 'Order details sent to customer via REST API.', 'woocommerce' ), false, true );
```

Comment:
I wonder if we could make this a bit more clear for non-technical users that the details were sent _in an email_, but the email itself was triggered by a REST API request. Or better yet, triggered by the user agent that sent the REST API request, since "REST API" probably doesn't mean anything to most people but a user agent string would show, e.g. that it came from one of the Woo mobile apps.

For the first part, including the actual email address would be a good indication, and that might be useful for the merchant anyway, in case they need to troubleshoot why a customer isn't getting the details.

So something like:

```
$note = sprintf(
	esc_html__( 'Order details sent to %1$s, via %2$s', 'woocommerce' ),
	esc_html( $order->get_billing_email() ),
	esc_html( {method to get user agent from request headers, fallback to "REST API"} )
);

$order->add_order_note( $note, false, true );
```

Reply:
Done! 👌